### PR TITLE
[openvino] ported upstream patches to 2023.1.0

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
**Changes requested here https://github.com/conan-io/conan-center-index/pull/19681#discussion_r1339945988:**
- Added upper bound for cmake tool
- Pinned scons tool to exact version

**Several patches from upstream to make the recipe itself more future proof and minimize conditions when adding newer versions:**
- Ported patch to enable GPU plugin under clang
- Ported patch to compile and work with GPU plugin statically
- Used `rapidjson` as a requirement instead of "local" copy of this dependency in 3rdparty. This allows to prevent possible ODR violations. 

**Common changes:**
- Introduced dependency files to make updates for newer OpenVINO versions more simpler - no needs to write conditional code in the recipe itself, because different dependency versions can be hidden in dedicated files.

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/openvino/all/conandata.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 sources:
   "2023.1.0":
     "openvino":
-      url: "https://github.com/openvinotoolkit/openvino/archive/47b736f63edda256d66e2bbb572f42a9d6549f6e.tar.gz"
-      sha256: "8a1dd5b35fe4b60440e97ee61e12e01aa4dd78b5be30550aa08fcd96c416e1d7"
+      url: "https://github.com/openvinotoolkit/openvino/archive/refs/tags/2023.1.0.tar.gz"
+      sha256: "ff88596b342440185874ddbe22874b47ad7b923f14671921af760b15c98aacd6"
```
Comment (ilya-lavrenov) [inline recipes/openvino/all/conandata.yml:5]: commit hash is the same, just replace with tag instead if commit hash:
```
$ git log -1 2023.1.0
commit 47b736f63edda256d66e2bbb572f42a9d6549f6e (tag: 2023.1.0)
Author: Tatiana Savina <tatiana.savina@intel.com>
Date:   Mon Sep 4 15:04:36 2023 +0200

    update links (#19563)
```

**Comment by ilya-lavrenov** [comment]
@SpaceIm @RubenRBS 
Could you please have a look? The comments from original PR were addressed here

**Comment by RubenRBS** [comment]
@ilya-lavrenov yup, it's in the review backlog :) Thanks for your patience

**Review by RubenRBS — APPROVED** [review]
We usually prefer to have a mapping in the recipe code itself, but I guess this also works! Thanks a lot :)

**Comment by ilya-lavrenov** [comment]
> We usually prefer to have a mapping in the recipe code itself, but I guess this also works! Thanks a lot :)

I supposed that if we have mapping in code - we need to change it with every version of OpenVINO. So, when we introduce new version - Conan CI will rebuild all other OpenVINO versions, while they are actually not touched.

I hoped to WA this rebuild issue with external files for dependencies, because OpenVINO rebuild is not a 2 mins task :)

**Comment by ilya-lavrenov** [comment]
Hi @SSE4 @theirix @ericLemanissier @jwillikers 
Could you please review?

**Review by jwillikers — APPROVED** [review]

**Pull request merged**